target/linux/octeon/base-files/lib/preinit/79_move_config

```diff
@@ -1,21 +1,28 @@
-# Copyright (C) 2014 OpenWrt.org
+#!/bin/sh
 
 move_config() {
 	. /lib/functions.sh
-	. /lib/upgrade/common.sh
-
-	case "$(board_name)" in
+	local board="$(board_name)"
+	local device
+	case "${board}" in
 		erlite)
-			mount -t vfat /dev/sda1 /mnt
-			[ -f "/mnt/$BACKUP_FILE" ] && mv -f "/mnt/$BACKUP_FILE" /
-			umount /mnt
+			device="/dev/sda1"
 			;;
 		itus,shield-router)
-			mount -t vfat /dev/mmcblk1p1 /mnt
-			[ -f "/mnt/$BACKUP_FILE" ] && mv -f "/mnt/$BACKUP_FILE" /
-			umount /mnt
+			device="/dev/mmcblk1p1"
 			;;
 	esac
+
+	# checks for:
+	# 1. device variable is not empty
+	# 2. device variable points to actual block device
+	if [ ! -z "${device}" ] && [ -b "${device}" ] ; then
```

Comment:
! -z is the same as -n. Run through shellcheck.

Reply:
`-n` checks if variable exists. even empty.
`! -z` will also tell if it is not empty.

Reply:
https://github.com/koalaman/shellcheck/wiki/SC2236

Reply:
thank you.

pushed a fix resolving that.

also did `-n` with `return 1` on failure.
same thing but makes more sense in that case. also updated comments about that

Reply:
> `-n` checks if variable exists. even empty.

That's _wrong. `-n` will be _false_ for an empty string, no matter whether the variable exists or not.